presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -594,6 +595,9 @@ private static InnerJoinPushDownResult processInnerJoin(Expression inheritedPred
             leftEffectivePredicate = filterDeterministicConjuncts(leftEffectivePredicate);
             rightEffectivePredicate = filterDeterministicConjuncts(rightEffectivePredicate);
 
+            leftEffectivePredicate = stripDynamicFilters(leftEffectivePredicate);
```

Comment:
Instead add stripping to `EffectivePredicateExtractor.Visitor#visitFilter`.

Reply:
I'm just wondering. If we had dynamic filter implemented as a function, and if we had a function with that name bound we can avoid hacking around such places.

Reply:
This is removed and is not required anymore.